scripts/common/common.sh

```diff
@@ -1036,3 +1036,47 @@ function canonical_image_name() {
     fi
     echo "$image"
 }
+
+# wait_for_running_pods waits for pod(s) in a given namspace to transition to Running status
+function wait_for_running_pods() {
+    local namespace=$1
+    local is_job_controller=0
+    local ns_pods=
+    local status=
+    local containers=
+
+    ns_pods=$(kubectl get pods -n "$namespace" -o jsonpath='{.items[*].metadata.name}')
+
+    if [ -z "$ns_pods" ]; then
+        log "No pods found in namespace $namespace."
+        return 0
+    fi
+
+    for pod in $ns_pods; do
+        status=$(kubectl get pod "$pod" -n "$namespace" -o jsonpath='{.status.phase}')
+
+        # determine if pod is manged by a Job
+        if kubectl get pod "$pod" -n "$namespace" -o jsonpath='{.metadata.ownerReferences[*].kind}' | grep -q "Job"; then
+            is_job_controller=1
+        fi
+
+        if [ "$status" != "Running" ] && [ "$status" != "Succeeded" ]; then
```

Comment:
Should we not also add Completed?

Reply:
`Completed` pods have a status.phase of `Succeeded`.

Reply:
We are testing this one here : https://testgrid.kurl.sh/run/pr-4101-5175666-rook-1.10.11-k8s-docker-2023-02-14T22:34:55Z Am I right? So let's just wait the tests to move forward, otherwise /lgtm